File: sms/src/main/java/dev/pitlor/sms/Contacts.java
package dev.pitlor.sms;

import java.util.List;

import javax.inject.Inject;

import dev.pitlor.sms.repositories.ContactRepository;

public class Contacts {
    private ContactRepository contactRepository;

    @Inject
    public Contacts(ContactRepository contactRepository) {
        this.contactRepository = contactRepository;
    }

    public List<Contact> readAll(int limit) {
        return null;
    }
}

File: sms/src/main/java/dev/pitlor/sms/Contacts.kt
package dev.pitlor.sms

import dev.pitlor.sms.repositories.ContactRepository
import javax.inject.Inject

class Contacts @Inject constructor(private val contactRepository: ContactRepository) {
    fun readAll(numbers: List<String>): List<Contact> {
        return numbers.map(contactRepository::getContact)
    }
}